### ISSUE TITLE
Question - recovery code neither works nor completes

### ISSUE THREAD

**Issue by Phosra** [oxen-io/session-android#1008]
After leaving my account for a few months (not on any device), I returned with the recovery code, yet am finding that it slightly dims the screen, yet never succeeds in reconnecting with the account.
I will be prompted after a minute, offering to skip the process, but I continue to wait up to even ten minutes, progressing to trying mobile data connections, different Wi-Fi connections, ToR bridges through Orbot, etc. without a different result.

Why might this occur? Is this a bug?
Are there any debug logs that may be useful for the team here? (Without exposing my personal data, of course.)

**Comment by mpretty-cyro** [comment]
So it looks like on Android when restoring a device we wait until we receive a "configuration message" (ie. cached user preferences) until we continue past this screen but the configuration message has the same 14 day server expiration as all other messages which are send via Session - in your case because you haven't used your account for a few months it no longer exists in the swarm which is why this restoration is failing.

@hjubb We could probably look at updating the logic so we also update the state if the poller returns `"more": false` before receiving a configuration message which would help prevent this situation?

The other option I can think of is updating `TextSecurePreferences.CONFIGURATION_SYNCED` based on the above flag but that seems like it would be technically incorrect and could lead to unexpected behaviours down the track

**Comment by hjubb** [comment]
@mpretty-cyro yes, however if we skip the wait for configuration sync message it will see that we haven't synced upon entering the app and trigger a new config to be sent iirc. It should mean that the config message synced flag will be set after that point anyway I think. Would have to check the logic (it should be that case for creating a new session ID anyway I think)

**Comment by mpretty-cyro** [comment]
I'm pretty sure that's actually how iOS behaves at the moment (ie. sends a new, essentially blank, config message immediately - would need to double check though)

Sounds like making use of the `"more": false` flag would be the best option here though - this would also continue to work down the line when user config messages are split into their own namespace

**Comment by Phosra** [comment]
Just to confirm, this sounds like a software bug here, yes? That the account should be accessible otherwise?

**Comment by KeeJef** [comment]
Yeah this is a bug and something we will be fixing when we tackle onboarding, basically we will wait for a user configuration message to be found but after a timeout we will allow the user to restore without one

**Comment by Phosra** [comment]
Is there any estimate of when onboarding is going to be worked on, to gauge when I will take a peek around here again?

**Comment by KeeJef** [comment]
We will be releasing fixes shortly alongside our update of user configuration messages
